internal/primitives/runtime/interfaces.go

```diff
@@ -95,13 +95,13 @@ type Header[N Number, H Hash] interface {
 // Block represents a block. It has types for Extrinsic pieces of information as well as a Header.
 //
 // You can iterate over each of the Extrinsics and retrieve the Header.
-type Block[N Number, H Hash] interface {
+type Block[N Number, H Hash, E Extrinsic] interface {
```

Comment:
Is there a reason why we need to supply generic param here? I was thinking about it too when introducing `Block` but didn't see a reason at the time.

Reply:
> Yes I needed it to be consistent with [this change](https://github.com/ChainSafe/gossamer/pull/4542/files#diff-3041859536678ec5e1b6ba9ecf62af97a304beccabe0a71656599b2fd3c2bf78R53)

Is there a need to specify a specific `runtime.Extrinsic` type when constructing a `Block` through `NewBlock`?